simpleviewinc/tap-events-force
# ZEN-669 waitlist master hotfix

**Ticket**: [ZEN-669](https://simpleviewtools.atlassian.net/browse/zen-669)

## Goal

* Add support for waitlist capacities in the group booking modal

## Updates
* Implemented the waitlist capacity, including adding UI to show when the waitlist is full, and disabling attendees when it is full
* Wrote extensive integration tests for it and updated one unit test
* Added some docs

## Testing

* `keg evf pack run package=evf:zen-669-waitlist-master-hotfix`
* Navigate to http://evf-zen-669-waitlist-master-hotfix.local.keghub.io/?state=waitingList-group-waitCapacity
* Open the group booking modal of the session
* Verify that it says there are `0 places remaining` (because the regular `capacity` property is 0)
* Next, try adding another person to the waiting list
* Verify that you see the status text "Waiting list full" at the top right of the modal
* Also verify that the remaining unchecked attendee checkboxes are now disabled (because neither the waiting list nor the booking list have any more capacity available). However, the checked boxes should still be enabled!
* Verify that the status text reads "1 place remaining" when you remove a user from the booking list
* Verify that the status text reads "0 places remaining" when you remove a user from the waiting list (after having previously filled it to capacity)
